### PR TITLE
[flutter_tools] add feature for iOS, android, and fuchsia

### LINKED ISSUE #58999: Adding 'flutter config' for mobile
Today, we have configuration options for pre-stable platform support:

```sh
$ flutter config --[no-]enable-web
$ flutter config --[no-]enable-macos-desktop
$ flutter config --[no-]enable-windows-desktop
$ flutter config --[no-]enable-linux-desktop
```

The flutter doctor command uses this information to provide per-platform diagnostics information. However, we don’t have the mobile equivalents:

```sh
$ flutter config --[no-]enable-android
$ flutter config --[no-]enable-ios
```

The consequence of this is that flutter doctor has no way to know whether a Flutter developer is ever going to use their machine for mobile development, so assumes that they are and provides all kinds of scary output directing the user to set up tools and libraries that they may never actually need.

By adding the mobile config options, we enable flutter doctor to tailor it’s output so as to avoid asking developers to do setup work that they do not need to do.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Add feature flags for android, ios, and fuchsia (on by default). After updating the g3 rollers, the fuchsia feature will be turned off by default. Creates a simpler base type of feature flags for g3 to extend.

Updates android, ios, fuchsia workflows to use feature flags check.
Removes concept of stable artifacts and checks on flutter version.

Fixes https://github.com/flutter/flutter/issues/58999
https://github.com/flutter/flutter/issues/52859
https://github.com/flutter/flutter/issues/12768

### CODE REVIEW

**Review by jmagman — APPROVED** [review]
> After updating the g3 rollers, the fuchsia feature will be turned off by default. 

I'm guessing this PR has to roll, then you'll immediately update to make fuchsia off?  Definitely don't want our users to start by default caching fuchsia artifacts.

File: packages/flutter_tools/lib/src/features.dart [hunk 1]
```diff
@@ -19,27 +19,66 @@ FeatureFlags get featureFlags => context.get<FeatureFlags>();
...
-  bool get isWebEnabled => isEnabled(flutterWebFeature);
+  bool get isWebEnabled => false;
 
   /// Whether flutter desktop for Windows is enabled.
-  bool get isWindowsEnabled => isEnabled(flutterWindowsDesktopFeature);
+  bool get isWindowsEnabled => false;
 
   /// Whether fast single widget reloads are enabled.
-  bool get isSingleWidgetReloadEnabled => isEnabled(singleWidgetReload);
+  bool get isSingleWidgetReloadEnabled => false;
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/features.dart]: Done!

**Comment by jonahwilliams** [comment]
> I'm guessing this PR has to roll, then you'll immediately update to make fuchsia off? Definitely don't want our users to start by default caching fuchsia artifacts.

Not quite - we still won't download fuchsia artifacts by default if precache is called, only android and iOS have that behavior. The feature is also disabled on dev, beta, stable channels so it isn't exposed any more than it was before

**Comment by jmagman** [comment]
```
google3:///mobile/flutter/tools/lib/context.dart:798:9: Error: A constant constructor can't call a non-constant super constructor.
  const _Google3Features();
```

**Comment by jonahwilliams** [comment]
noooooo

**Pull request merged**